docs/Simorgh-Overview.md

```diff
@@ -0,0 +1,75 @@
+
+Now that the raw HTML has been downloaded the Client.JS file kicks in and hydrates the initial response with the client side application. During this process reacts uses the initial JSON payload (available on the global window object `SIMORGH_DATA`) to hydrate the original markup returned by ReactDOMServer. React expects that the rendered content is identical between the server and the client (This is why we send the initial JSON payload with the SSR page, so the hydration phase runs with the same data that the server render used.)
+
+### Onward Journeys - Client Side Render (CSR)
+
+Now that the SPA is bootstrapped on the users device and onward journeys to further Simorgh pages are captured by react-router and rendered on entirely on the client.
+
+Clicking on an onward Journey triggers the `useEffect` hook in the main App container as the url path has changed. This effect uses `getRouteProps` again to match the new path against a react-router route, gaining access to the; service and isAmp props, the react container to render and the `getInitialData` function.
+
+The local state is now updated setting `loading` to true and data to null. This update to state triggers the App container to re-render passing the state values as props to the main container in this case either ArticlesContainer or FrontPageContainer. These containers compose a set of higher order components (HOC) and each one handles the values from a given props. Currently the loading prop is set to true so the `withLoading` HOC will return the loading component showing a visual loading state to the user.
+
+During this time the App container has finished fetching the JSON payload for the onward journey and the state is updated again. Loading is now set to false and data is set to the returned JSON data.
+
+The routes container is rendered again, this time loading is set to false so the `withData` HOC can run some validation against the JSON payload then return either an error components of the original page container e.g. ArticleContainer passing in the JSON data with the pageData prop.
+
+## Rendering a Page
+
+The JSON payload for an article consists of a number of Blocks. Each block an object which represents an element on the page, this could be a Heading, an Image, a Paragraph etc. each of these blocks has a block type and a block type will match up to a specific container in Simorgh e.g. blockType: image will match to the Image container.
```

Comment:
```suggestion
The JSON payload for an article consists of a number of Blocks. Each block is an object which represents an element on the page, this could be a Heading, an Image, a Paragraph etc. Each of these blocks has a block type and a block type will match up to a specific container in Simorgh e.g. blockType: image will match to the Image container.
```